Callout ECR not being supported (#984)
astronomer/docs: astro/kubernetespodoperator.md

By default, the KubernetesPodOperator expects to pull a Docker image that's hosted publicly on Docker Hub. If you want to execute a Docker image that's hosted in a private registry, complete the setup below.

:::caution

> Amazon Elastic Container Registry (ECR) is not supported. ECR credentials expire after 12 hours and they can't be recreated.

Reviewer: ```suggestion
Running Docker images that are hosted on [Amazon Elastic Container Registry](https://aws.amazon.com/ecr/) (ECR) is not supported on Astro. Authentication tokens generated by ECR expire after 12 hours and cannot be recreated, which means that KubernetesPodOperator tasks would not reliably have the necessary credentials to pull and run any Docker image hosted on ECR.

If you're using ECR and are interested in exploring other solutions, contact [Astronomer support](https://support.astronomer.io).
```

> AWS Elastic Container Registry (ECR) is currently not supported on Astro. ECR credentials expire after 12 hours and there is no way for Astro to create a new set of credentials.

Reviewer 2: ```suggestion
Amazon Elastic Container Registry (ECR) is not supported. ECR credentials expire after 12 hours and they can't be recreated.
```

By default, the KubernetesPodOperator expects to pull a Docker image that's hosted publicly on Docker Hub. If you want to execute a Docker image that's hosted in a private registry, complete the setup below.

:::caution

> Running Docker images that are hosted on [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, KubernetesPodOperator tasks don't have the credentials necessary to pull and run Docker images hosted on ECR.

Author: ```suggestion
Running Docker images that are hosted on [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, Kubelet will not have the credentials necessary to pull and run Docker images hosted on ECR in KubernetesPodOperator tasks.
```

Author: Technically this is more correct but might confuse people

> Running Docker images that are hosted on [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, KubernetesPodOperator tasks don't have the credentials necessary to pull and run Docker images hosted on ECR.

Reviewer: ```suggestion
Running Docker images that are hosted on [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, KubernetesPodOperator tasks don't have access to the credentials necessary to pull and run Docker images hosted on ECR.
```

Yea, I think that's too low-level -- @stuart23 this better?

Author: The intent is ok. I don't think people are going to pull us up on it, so I don't mind if you want to keep it more readable.

Reviewer 2: Maybe:

"...can't access the credentials necessary..."

> Running Docker images that are hosted on [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, KubernetesPodOperator tasks cannot have the credentials necessary to pull and run Docker images hosted on ECR.
> 
> If you're using ECR and are interested in exploring other solutions, contact [Astronomer support](https://support.astronomer.io).

Reviewer: ```suggestion
Running Docker images that are hosted on [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, KubernetesPodOperator tasks don't have the credentials necessary to pull and run Docker images hosted on ECR.

If you're using ECR and are interested in exploring other solutions, contact [Astronomer support](https://support.astronomer.io).
```

Author: Technically the first option is more correct

> Running Docker images that are hosted on [Amazon Elastic Container Registry](https://aws.amazon.com/ecr/) (ECR) is not supported on Astro. Authentication tokens generated by ECR expire after 12 hours and cannot be recreated, which means that KubernetesPodOperator tasks would not reliably have the necessary credentials to pull and run any Docker image hosted on ECR.
> 
> If you're using ECR and are interested in exploring other solutions, contact [Astronomer support](https://support.astronomer.io).

Reviewer: ```suggestion
Running Docker images that are hosted on [Amazon Elastic Container Registry](https://aws.amazon.com/ecr/) (ECR) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, KubernetesPodOperator tasks cannot have the credentials necessary to pull and run Docker images hosted on ECR.

If you're using ECR and are interested in exploring other solutions, contact [Astronomer support](https://support.astronomer.io).
```
